### DISCUSSION TITLE
Feature request: Can disable scrolling ScrollArea by dragging.

### DISCUSSION THREAD

**Discussion by ruimo** [emilk/egui#1576] [Ideas]
Hi, current ScrollArea responds the dragging that scroll its content. This is reasonable for touch devices but for Desktop application (using mice), it is typically an action for area selection (drawing dashed box) or moving objects around within the ScrollArea. The current scrolling_enabled is effective both dragging scroll and scrollbars. I suggest to add a property to enable/disable scroll by dragging. Or add some kind of flavor that expresses if the device is touch device.

**Comment by emilk** [comment]
Perhaps it would be sensible to disable "drag to scroll" on non-touch devices.

Still: if you fill a `ScrollArea` with things that has `Sense::drag`, then these things will be dragged rather than the scroll area, so the `ScrollArea` should never "steal" drags from anything else that are interested in them (e.g. sliders).

**Comment by ruimo** [comment]
Thanks! Calling Ui::interact() with Sense::drag(), I can let ScrollArea ignore the scrolling by dragging.
